include/deal.II/base/index_set.h

```diff
@@ -59,6 +59,145 @@ DEAL_II_NAMESPACE_OPEN
 class IndexSet
 {
 public:
+
+  typedef types::global_dof_index index_type;
```

Comment:
Conceptually, the type you want here is probably something so that you can write
```
  index_set[index]
```
when determining whether `index` is an element of the set. The STL uses the name `size_type` for such types.